docs/tutorials/google/advanced_calibration.ipynb

```diff
@@ -0,0 +1,799 @@
+{
```

Comment:
<p>I would be against doing characterizations together: they take a long time and the fresher the results are, the better resulting quality.</p><p> </p><p>After each characterization the corresponding calibrated circuits should be executed immediately. The code here should be merged with the code below with loop over options being the outermost one.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/quantumlib/Cirq/pull/4108/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='38'/>

Reply:
<p>Good feedback, thanks! Perhaps this will make the results closer to what is expected...</p>

Reply:
<p>I would split these up so folks following along can pick one or the other to copy-paste without having to disentangle a for loop</p>

Reply:
<p>Agree. It's split up now and should be easy to skip/comment out either calibration.</p><p> </p><p>I still do both calibrations then run Loschmidt instead of [calibrate, Loschmidt] x 2. Results are looking much better now I think, so maybe this is fine. Let me know what you think @mrwojtek. </p>

Reply:
<p>I see what you've been trying to do. I'll do some adjustments to the calibration methods to make your workflow easier.</p><p> </p><p>However, I propose some refactor which will simplify things significantly in this tutorial. It will also probably improve the performance (I expect that Floquet calibration can be improved if ran separately):</p><p> </p><ol><li>"Floquet calibration" section should just have the first cell, the one which defines floquet_options. Remove the second cell.</li><li>"XEB calibration" section should just have the first cell, the one which defines xeboptions. Remove the second cell.</li><li>In the "Run Loschmidt echo benchmark" you can just define:</li></ol><p>calibrations = {</p><p>&nbsp;&nbsp;"No calibration": None,&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</p><p>&nbsp;&nbsp;"Floquet calibration": floquet_options,</p><p>&nbsp;&nbsp;"XEB calibration": xeb_options`</p><pre class="ql-syntax" spellcheck="false">}
</pre><p> </p><p>Then you just need to do:</p><p>batch_to_compensate, characterization_requests = cg.prepare_characterization_for_circuits_moments(loschmidt_echo_batch, options)</p><p> </p><p>characterizations = cg.run_calibrations(</p><p>&nbsp;&nbsp;characterization_requests,&nbsp;</p><p>&nbsp;&nbsp;sampler,</p><p>)</p><p> </p><p>batch = [</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;cg.make_zeta_chi_gamma_compensation_for_moments(circ, characterizations).circuit</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;for circ in batch_to_calibrate</p><p>&nbsp;&nbsp;&nbsp;&nbsp;]</p><p> </p><p>results = sampler.run_batch(programs=batch, repetitions=nreps)</p>

Reply:
<p>This is now done, with a note to update after https://github.com/quantumlib/Cirq/pull/4118 is merged. </p>

Reply:
<p>I also added another calibration for the user-defined circuit, instead of reusing one of the characterizations from the Loschmidt echo, as we discussed offline. </p>

Reply:
@rmlarose I don't see the changes mentioned here in this document? Did you forget to push your changes?

Reply:
Just to document it here, they were in https://github.com/quantumlib/Cirq/pull/4108/commits/dfbe326d768e883d8eb9d3be36adfda586b22351 which was reverted.